flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1395,13 +1417,12 @@ mlir::SymbolRefAttr IntrinsicLibrary::getUnrestrictedIntrinsicSymbolRefAttr(
   // this before calling the code generators.
   bool loadRefArguments = true;
   mlir::FuncOp funcOp;
-  for (const auto &handler : handlers)
-    if (name == handler.name)
-      funcOp = std::visit(
-          [&](auto generator) {
-            return getWrapper(generator, name, signature, loadRefArguments);
-          },
-          handler.generator);
+  if (auto *handler = findIntrinsicHandler(name))
+    funcOp = std::visit(
+        [&](auto generator) {
```

Comment:
The previous `std:::visit` pattern used `auto &`. Should one use a reference as well? For that matter, should they all be `const auto &`?

Reply:
I'll switch to `const auto &`.